### ISSUE TITLE
Move non-core built-ins to optional libraries

### ISSUE THREAD

**Issue by jathak** [Cal-CS-61A-Staff/dart_scheme#83]
While it's not as bad as the [old interpreter](https://scheme-legacy.apps.cs61a.org) in this regard, there's a decent chunk of built-in procedures that aren't really relevant to this interpreter's core purpose: teaching 61A students Scheme.

To that end, I'd like to restrict the core library to the following:

- Core functionality implemented in the project and described in the [61A Scheme specification](https://cs61a.org/articles/scheme-spec.html)
- Core learning and teaching tools that course staff show students (or use themselves)
  - Mainly the diagramming and documentation procedures
- Core quality-of-life utilities that are frequently used
  - `clear`, `bindings`, `theme`, maybe a few others

For everything else, we'll move them into a set of core libraries that can be imported when needed, similar to how the Logic library currently works (though we'll only have a single `library` procedure for imports, rather than a separate loading procedure for each library).

tldr: Does it *~~spark joy~~* teach Scheme? If not, throw it in an optional library.
